app/models/slot_constraints.rb

```diff
@@ -0,0 +1,33 @@
+class SlotConstraints
+  delegate :map, to: :@slot_calendars
```

Comment:
`slot_calendars` does not make sense, I think it should follow the class name: `calendar_slots`